src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1224,6 +1224,27 @@ public Task<bool> HasSuccessfullyLoadedAsync(ProjectState project, CancellationT
             : project.HasAllInformation ? SpecializedTasks.True : SpecializedTasks.False;
     }
 
+    public SolutionCompilationState WithForceOnlyRequiredGenerators(ProjectId projectId)
+    {
+        if (!_projectIdToTrackerMap.TryGetValue(projectId, out var tracker))
+        {
+            // we don't have a tracker for this yet, create one with the correct policy
+            tracker = RoslynImmutableInterlocked.GetOrAdd(
+                ref _projectIdToTrackerMap,
+                projectId,
+                static (id, solutionState) => CreateCompilationTracker(id, solutionState).WithCreateOnlyRequiredGeneratorDocs_DoNotCreateSkeletonReferencesCreationPolicy(),
```

Comment:
The method name here WithForceOnlyRequiredGenerators implies we're updating generator state, but we're also implicitly changing the skeleton references policy here too? Does that matter?